lxrhash.go

```diff
@@ -93,13 +92,10 @@ func (lx *LXRHash) Hash(src []byte) []byte {
 
 	idx = 0
 	// Actual work to compute the hash
-	for i, v2 := range src {
+	for _, v2 := range src {
 		if idx >= lx.HashSize { // Use an if to avoid modulo math
 			idx = 0
 		}
-		if i == 0 {
-			lx.FirstIdx = (as>>5 ^ uint64(v2)) & mk
```

Comment:
Adam Levy suggested removing the pointer receiver from the Hash function.  It was needed so that we _could set lx.FirstIdx_.  Since we are not, we can remove it.

Does add copying that we are now avoiding, but I doubt the difference is measurable.

Reply:
Sure, removed.